ote_sdk/ote_sdk/tests/entities/shapes/test_ellipse.py

```diff
@@ -0,0 +1,318 @@
+        }
+
+        with pytest.raises(ValueError):
+            Ellipse(**incorrect_ellipse_params)
+
+        width_lt_height_ellipse_params = {
+            "x1": 0.4,
+            "x2": 0.5,
+            "y1": 0.3,
+            "y2": 0.4,
+        }
+
+        width_lt_height_ellipse = Ellipse(**width_lt_height_ellipse_params)
+        assert width_lt_height_ellipse.height > width_lt_height_ellipse.width
+        assert width_lt_height_ellipse.major_axis == width_lt_height_ellipse.height / 2
+        assert width_lt_height_ellipse.width == 0.09999999999999998
+        assert width_lt_height_ellipse.height == 0.10000000000000003
+        assert width_lt_height_ellipse.x_center == 0.45
+        assert width_lt_height_ellipse.y_center == 0.35
+        assert width_lt_height_ellipse.minor_axis == 0.04999999999999999
+        assert width_lt_height_ellipse.major_axis == 0.05000000000000002
+
+        width_gt_height_ellipse = Ellipse(**self.width_gt_height_ellipse_params())
+        assert width_gt_height_ellipse.height < width_gt_height_ellipse.width
+        assert width_gt_height_ellipse.minor_axis == width_gt_height_ellipse.height / 2
+        assert width_gt_height_ellipse.width == 0.30000000000000004
+        assert width_gt_height_ellipse.height == 0.19999999999999998
+        assert width_gt_height_ellipse.x_center == 0.65
+        assert width_gt_height_ellipse.y_center == 0.2
+        assert width_gt_height_ellipse.minor_axis == 0.09999999999999999
+        assert width_gt_height_ellipse.major_axis == 0.15000000000000002
+
+    @pytest.mark.priority_medium
+    @pytest.mark.component
+    @pytest.mark.reqids(Requirements.REQ_1)
+    def test_ellipse_magic_methods(self):
+        """
+        <b>Description:</b>
+        Check Ellipse __repr__, __eq__, __hash__ methods
+
+        <b>Input data:</b>
+        Initialized instance of Ellipse
+
+        <b>Expected results:</b>
+        Test passes if Ellipse magic methods returns correct values
+
+        <b>Steps</b>
+        1. Initialize Ellipse instance
+        2. Check returning value of magic methods
+        """
+
+        x1 = self.ellipse_params()["x1"]
+        x2 = self.ellipse_params()["x2"]
+        y1 = self.ellipse_params()["y1"]
+        y2 = self.ellipse_params()["y2"]
+
+        ellipse = self.ellipse()
+        assert repr(ellipse) == f"Ellipse(x1={x1}, y1={y1}, x2={x2}, y2={y2})"
+
+        other_ellipse_params = {
+            "x1": 0.5,
+            "x2": 1.0,
+            "y1": 0.0,
+            "y2": 0.5,
+            "modification_date": now().replace(microsecond=False),
+        }
+
+        third_ellipse_params = {
+            "x1": 0.3,
+            "y1": 0.5,
+            "x2": 0.4,
+            "y2": 0.6,
+            "modification_date": now().replace(microsecond=False),
+        }
+
+        other_ellipse = Ellipse(**other_ellipse_params)
+        third_ellipse = Ellipse(**third_ellipse_params)
+
+        assert ellipse == other_ellipse
```

Comment:
I guess that it can be false in some rare cases even if you replace microseconds by 0. 
Image that SECOND (but not MICROSECOND) was changed between two calls of `now`.

I would call `now` only once, store it as class member and re-use where it is needed, without replacing microseconds.

Reply:
Sure. Reworked.